src/pages/errorpages/Forbidden.tsx

```diff
@@ -1,26 +1,42 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { Link } from 'react-router-dom';
-import { Link as MuiLink } from '@material-ui/core';
+import { Link as MuiLink, CircularProgress } from '@material-ui/core';
 import styled from 'styled-components';
-
 import Heading from '../../components/Heading';
 import NormalText from '../../components/NormalText';
+import { StyledProgressWrapper } from '../../components/styled/Wrappers';
+import { LOAD_ERROR_PAGE_AFTER_DURATION } from '../../utils/constants';
 
 const StyledLinkText = styled(NormalText)`
   margin-top: 1rem;
 `;
 
 const Forbidden: FC = () => {
   const { t } = useTranslation('authorization');
+  const [showSpinner, setShowSpinner] = useState(true);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setShowSpinner(false);
+    }, [LOAD_ERROR_PAGE_AFTER_DURATION]);
+  }, []);
```

Comment:
Vet du hvorfor det var et problem tidligere? Det er litt synd å ha en fixed 3sec wait hvis det egentlig bare er gaffateip over en mulig bug

Jeg mener egentlig at sjekken i `App.tsx` skal vise spinner helt til `user` er lastet (evt. mangler, ergo ikke logget inn), og først etter det skal alt annet rendres (som inkluderer dette). Dermed skal vi vite at vi enten har en innlogget bruker, eller en ikke innlogget bruker når vi kommer hit. Hvis vi her må vente på au user er lastet igjen tyder det kanskje på at det er en bug for loading state i `App.tsx`?

Reply:
mulig det er dette som klusser til, siden `getCurrentUserAttributes()` er async...
```
const getUser = async () => {
      dispatch(getCurrentUserAttributes());
      setIsLoadingUser(false);
    };
```

I så fall kan sikkert dette gi problemer flere steder også :/

Reply:
tror `getCurrentUserAttributes` burde vært et vanlig async api-kall, så venta vi på respons, dispatcha det, _så_ satt loading til false

Reply:
Bugen her er at når man er inn på en side som krever innlogging (f.eks "Min profil"), og logger ut, så flashes `forbidden` før man faktisk blir logget ut. Det som er problemet er at routingen skjer før hentingen av `user`. Så hvis vi skal unngå å flashe `forbidden`, så må vi vente.

Reply:
Det er forsåvidt kanskje bedre å sette `isLoadingUser` i reduceren til "SET_USER" action'en når jeg tenker meg om. Prøver det først.

Reply:
ah, kom borti lignende ved innlogging en gang. Mulig vi bare kan fjerne `signOut`-casen fra `hub-listener.ts`. Tror den trigges før man blir redirected til URL som er satt i env, så den er kanskje overflødig(?)

Reply:
Kan sjekke 👍

Reply:
Tror du har rett der ja! 💯